cmd/libsnap-confine-private/error.c

```diff
@@ -81,6 +83,8 @@ sc_error *sc_error_init_api_misuse(const char *msgfmt, ...)
 
 const char *sc_error_domain(sc_error * err)
 {
+	// Set errno in case we die.
+	errno = 0;
 	if (err == NULL) {
 		die("cannot obtain error domain from NULL error");
```

Comment:
maybe it should be errno = EINVAL in this branch and likely in other places doing sanity checks? Though this would break the test, as they expect this particular error string, without the following `: Invalid argument`:thinking:

Reply:
Thanks, yeah, this PR tries to fix the issue in a minimal way but we can of course do a followup that does more changes.